### PR TITLE
Save in history incorrect REPL expressions

### PR DESCRIPTION
**Pull request by JesusValera**
## 📚 Description

Closes https://github.com/phel-lang/phel-lang/issues/205

When you press `<ENTER>` in the REPL, even if the statement is incorrect, it will be saved inside the `phel-repl-history` file.

## 🖼️  Screenshot

image

### CODE REVIEW

File: src/php/Command/ReplCommand.php [hunk 1]
```diff
@@ -101,6 +101,14 @@ private function addLineFromPromptToBuffer(): void
             $this->io->writeln();
         } else {
             $this->inputBuffer[] = $input;
+            $this->addHistory($input);
+        }
+    }
+
+    private function addHistory(?string $input): void
+    {
+        if (!empty($input)) {
```
Comment (Chemaclass) [inline src/php/Command/ReplCommand.php:110]:
```suggestion
    private function addHistory(string $input): void
    {
        if ('' !== $input)) {
```

**Review by Chemaclass — APPROVED** [review]
Good one!

**Review by jenshaase — CHANGES_REQUESTED** [review]
This is not working correctly in multi line mode. If I type this
```
phel:3> (
....:4> +
....:5> 1
....:6> 3
....:7> )
4
phel:8>
```
The history is `(`, `+`, `1`, `3`, `)` and not
```
(
+
1
3
)
```

**Comment by JesusValera** [comment]
> This is not working correctly in multi line mode. If I type this
> 
> ```
> phel:3> (
> ....:4> +
> ....:5> 1
> ....:6> 3
> ....:7> )
> 4
> phel:8>
> ```
> 
> The history is `(`, `+`, `1`, `3`, `)` and not
> 
> ```
> (
> +
> 1
> 3
> )
> ```

Do you mean, if you press `up`, it should display all those lines in a single one?

**Comment by jenshaase** [comment]
Not in a single row but like this
```
phel:3> (
....:4> +
....:5> 1
....:6> 3
....:7> )
4
phel:8> (
+
1
3
)
```

That's how it was before.

**Comment by JesusValera** [comment]
Sorry for the delay.

Basically, I was storing into the `history` line-by-line, the solution was storing just after imploding all the input buffer. 😄 

@jenshaase @Chemaclass

**Comment by Chemaclass** [comment]
You are storing the inputBuffer every time even when it was Unfinished.  Test this:
1) delete all content from the `.phel-repl-history`
2) exec in repl
```clojure
(defn adder [& xs]
  (apply + xs)
  ]
```
3) the `.phel-repl-history` should have only 1 line

**Comment by JesusValera** [comment]
> You are storing the inputBuffer every time even when it was Unfinished. Test this:
> 
> 1. delete all content from the `.phel-repl-history`
> 2. exec in repl
> 
> ```clojure
> (defn adder [& xs]
>   (apply + xs)
>   ]
> ```
> 
> 1. the `.phel-repl-history` should have only 1 line

That's true!
Thank you for spotting this bug :P

FIXED!

**Pull request merged**